components/kyma-environment-broker/internal/process/update/btp_operator_overrides.go

```diff
@@ -62,7 +61,7 @@ func (s *BTPOperatorOverridesStep) Run(operation internal.UpdatingOperation, log
 
 	// found btp-operator in last runtime state but config isn't matching
 	l := operation.LastRuntimeState.ClusterSetup.KymaConfig.Components[last]
-	if !reflect.DeepEqual(l, ci) {
```

Comment:
the deep equal actually wasn't reliable, we want to check specifically BTP operator overrides.